### PR TITLE
Fix: Timeout error while setting Mqtt Broker URL

### PR DESCRIPTION
**Pull request by nlrcomcast**
RDKB-56191: Observed CCSP Timeout error while setting Mqtt Broker URL

### CODE REVIEW

File: src/mqttcm_connect.c [hunk 1]
```diff
@@ -143,6 +159,14 @@ int mqttCMRbusInit(char *pComponentName)
 		return 0;
 	}
 	MqttCMInfo("mqttCMRbusInit is success. ret is %d\n", ret);
+
+
+	err = pthread_create(&disconnect_tid, NULL, disconnect_thread,NULL);
```
Comment (sadhyama) [inline src/mqttcm_connect.c:164]: Disconnect thread cannot be added in rbus init func.

Comment (nlrcomcast) [reply src/mqttcm_connect.c:169]: created function for thread

File: src/mqttcm_connect.c [hunk 2]
```diff
@@ -130,6 +133,19 @@ void mosquittoTriggerDisconnect()
 	valueChangeFlag = 0;
 }
 
+void *disconnect_thread(void *status) {
+	pthread_detach(pthread_self());
+	while (1)
+	{
+		pthread_mutex_lock(&disconnect_mutex);
+		pthread_cond_wait(&disconnect_cond, &disconnect_mutex);
```
Comment (sadhyama) [inline src/mqttcm_connect.c:141]: Add an info log here as "Received signal to disconnect , proceeding to disconnect with broker"

Comment (nlrcomcast) [reply src/mqttcm_connect.c:141]: added

**Pull request merged**